Adding support for the LSM303DLHC e-Compass

BryanSouza91 opened the pull request:
I have added support for the LSM303DLHC e-Compass using the existing LSM303AGR driver as the starting point. I updated the registers as necessary from the information provided by the datasheet. I also fixed spelling in the connection error message in the LSM303AGR driver.

BryanSouza91 committed:
added support for LSM303DLHC e-Compass;

BryanSouza91 committed:
fixed the spelling in the Connection error message; Initial support for LSM303DLHC added;

deadprogram commented:
Thanks for working on this @BryanSouza91 

Can you please add an example, and also then add that example to the smoke tests? https://github.com/tinygo-org/drivers/blob/dev/smoketest.sh

Thank you!

aykevl committed:
ws2812: add RP2350 support

Adding 150MHz support for the RP2350

ysoldak committed:
lsm6ds3tr: avoid unnecessary heap allocations (#766)

* lsm6ds3tr: avoid unnecessary heap allocations
* lsm6ds3tr: use helper functions, for readability
* lsm6ds3tr: return slice of the internal buffer on readBytes

jphastings committed:
feat: allow gps init with address

Adafruit's Mini GPS PA1010D Module works with this device driver, but requires 0x10 as the address, rather than 0x42.

This change allows the device to be initialised with whatever i2c address is needed, while maintaining backward compatibility.

Adds new constants to allow easy configuration of both the ublox device and the PA1010D.

ysoldak committed:
ssd1306: avoid unnecessary heap allocations (#767)

* ssd1306: avoid unnecessary heap allocations

* ssd1306: extract i2c and spi bus implementations

* ssd1306: refactor tests -- show fps and heap usage

* ssd1306: bring back the lost exported methods

* Adjust examples

* Fix smoketests for ssd1306

ksinistr committed:
Add ens160 i2c driver

Driver for ENS160 sensor:
https://www.sciosense.com/wp-content/uploads/2023/12/ENS160-Datasheet.pdf

BryanSouza91 committed:
Added LSM303DLHC to smoketest and added an example;

BryanSouza91 committed:
Merge branch 'dev' into dev

BryanSouza91 commented:
@deadprogram I finally got around to adding the example and then adding that to the smoketest.

deadprogram commented on lsm303dlhc/lsm303dlhc.go:
```diff
@@ -0,0 +1,232 @@
+	MagDataRate    uint8
+}
+
+// commented out "Connected" related lines since the DLHC sensor does not have the WHO_AM_I registers
```
In that case, these lines should be removed instead of commented.

BryanSouza91 replied:
I have removed the commented lines that were related to unused functions.

deadprogram commented on lsm303dlhc/lsm303dlhc.go:
```diff
@@ -0,0 +1,232 @@
+	}
+}
+
+// Connected returns whether both sensor on LSM303dlhc has been found.
```
Same for all the commented lines...

BryanSouza91 replied:
Removed in latest commit.

BryanSouza91 committed:
Removed unnecessary comments;

BryanSouza91 committed:
fixed format error;

BryanSouza91 committed:
Merge branch 'dev' into dev

deadprogram commented:
@BryanSouza91 due to having added the merge commits to this PR, it is not really possible to merge in its current state. Perhaps you can cherry-pick your commits related to the LSM303DLHC into a separate branch, squash them, and then either overwrite the branch in this PR, or create a new PR.

What do you think?

BryanSouza91 commented:
I shouldn't have an issue squashing then overwriting this branch, if I do I will create a new PR.

BryanSouza91 committed:
Merge branch 'dev' into dev

BryanSouza91 commented:
@deadprogram I don't think that last commit will work... I have [dev-PR branch](/BryanSouza91/drivers/tree/dev-PR) ready to go.

deadprogram commented:
Closing since was replaced by #783 thank you!
